components/frontend/src/header_footer/SettingsPanel.js

```diff
@@ -494,6 +494,7 @@ function ResetSettingsButton(
         visibleDetailsTabs
     }
 ) {
+    const metricsToHideDefault = atReportsOverview ? "all" : "none"
```

Comment:
Yes, well spotted.

Reply:
I guess the way to solve this, is to have `useURLSearchQuery` return an object with different fields. So instead of 

```javascript
const metricsToHideDefault = atReportsOverview ? "all" : "none"
const [metricsToHide, setMetricsToHide] = useURLSearchQuery("metrics_to_hide" + urlSearchQueryKeyPostfix, "string", metricsToHideDefault);
```

we would have something like:

```javascript
const metricsToHideDefault = atReportsOverview ? "all" : "none"
const metricsToHide = useURLSearchQuery("metrics_to_hide" + urlSearchQueryKeyPostfix, "string", metricsToHideDefault);
console.log(metricsToHide.value);  // Would print the value of the parameter, e.g. "all"
metricsToHide.set("none");  // Would change the value
console.log(metricsToHide.default);  // Would print the default value
```

This would also reduce the number of parameters we pass around considerably, as we now pass both `metricsToHide` and `setMetricsToHide`.

Reply:
New issue:  #7349